Make `rusk-vm` run on self-hosted CI server

**Describe what you want implemented** 
Rusk VM CI should run on our faster self-hosted CI server.

**Describe "Why" this is needed**
With the code coverage the tests are taking too much time.

**Describe alternatives you've considered**
N/A

**Additional context**

Currently it's failing because cannot install `kcov`:

```
Run sudo apt install -y kcov
/home/docker/actions-runner/_work/_temp/9fde21c9-a27a-4600-9156-7c36bef670ed.sh: line 1: sudo: command not found
Error: Process completed with exit code 127.
```

--- a/.github/workflows/dusk_ci.yml
+++ b/.github/workflows/dusk_ci.yml
@@ -8,7 +8,7 @@ name: Continuous integration
 jobs:
   analyze:
     name: Dusk Analyzer
-    runs-on: ubuntu-latest
+    runs-on: core
     steps:
       - uses: actions/checkout@v2
       - uses: actions-rs/toolchain@v1
@@ -33,7 +33,7 @@ jobs:
 
   test_nightly:
     name: Nightly tests
-    runs-on: ubuntu-latest
+    runs-on: core
     steps:
       - name: Checkout
         uses: actions/checkout@v2
@@ -45,9 +45,6 @@ jobs:
 
       - name: Install WASM32 target
         run: rustup target add wasm32-unknown-unknown
-
-      - name: Install kcov
-        run: sudo apt install -y kcov
 
       - name: Build test contracts
         run: make test-contracts
@@ -75,7 +72,7 @@ jobs:
 
   fmt:
     name: Rustfmt
-    runs-on: ubuntu-latest
+    runs-on: core
     steps:
       - uses: actions/checkout@v2
       - uses: actions-rs/toolchain@v1

